src/components/search/data/searchLoader.test.jsx

```diff
@@ -124,4 +143,74 @@ describe('searchLoader', () => {
       }),
     );
   });
+
+  it('Redirect learners whose enterprise has enabled one academy.', async () => {
+    extractEnterpriseCustomer.mockResolvedValue(enterpriseCustomerFactory({ enable_one_academy: true }));
+    const academiesQuery = queryAcademiesList(mockEnterpriseCustomer.uuid);
+
+    when(mockQueryClient.ensureQueryData).calledWith(
+      expect.objectContaining({
+        queryKey: academiesQuery.queryKey,
+      }),
+    ).mockResolvedValue(mockAcademies);
+    ensureAuthenticatedUser.mockResolvedValue({ userId: 3, username: 'test-user' });
+
+    const mockParams = { enterpriseSlug: mockEnterpriseCustomer.slug };
+    const searchLoader = ({ request }) => makeSearchLoader(mockQueryClient)({ params: mockParams, request });
+
+    renderWithRouterProvider({
+      path: `/${mockEnterpriseCustomer.slug}/search`,
```

Comment:
The `path` should define the route params, not using actual values. In this case: 

```jsx
path: '/:enterpriseSlug/search',
```

The expected `enterpriseSlug` param comes through because the `path` includes `:enterpriseSlug` and the `initialEntries` list contains the real/expected slug value:

```jsx
initialEntries: [`/${mockEnterpriseCustomer.slug}/search`],
```